File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        triangle = [[1]]
        if numRows == 1:
            return triangle
        
        triangle.append([1, 1])
        for i in range(3, numRows + 1):
            triangle.append([1])
            for j in range(1, i - 1):
                triangle[-1].append(triangle[-2][j - 1] + triangle[-2][j])
            triangle[-1].append(1)

        return triangle
    

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
/*
simple addition based solution
*/
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> triangle = {{1}};
        if (numRows == 1) {
            return triangle;
        }

        triangle.push_back({1, 1});
        for (int i = 3; i < numRows + 1; ++i) {
            triangle.push_back({1});
            for (int j = 1; j < i - 1; ++j) {
                triangle.back().push_back(triangle[triangle.size() - 2][j - 1] + triangle[triangle.size() - 2][j]);
            }
            triangle.back().push_back(1);
        }

        return triangle;
    }
};